packages/lit-html/src/directive.ts

```diff
@@ -16,8 +16,8 @@ export {
   PropertyPart,
 } from './lit-html.js';
 
-export interface DirectiveClass {
-  new (part: PartInfo): Directive;
+export interface DirectiveClass<RenderAs = unknown> {
```

Comment:
Why do you need this type parameter? You'd normally get the return type with `ReturnType<C['render']>`.

Is it so that you can feed in a type variable from an override type for `directive()`, like in Guard? If so, can you leave a comment?

Reply:
We can actually use `infer` (and similar tricks in the TypeScript type system) to get around needing this at all. I've updated the PR to not need any changes to the common directive types, with a little care, a directive can just be written to put enough information on the DirectiveResults it makes.